File: website/doctor-site/src/app/doctor-list/doctor.ts
export class Doctor {
  constructor(
    public doctor_id?: string,
    public firstName: string = "",
    public lastName: string ="",
    public email: string = "",
    public password?: string | undefined,
    public specialist: string = "",
    public branch: string = "",
    public photo: string = "",
    public fee: number = 0,
  ) {}
}

File: microservices/doctor-service/src/main/java/com/example/entity/Doctor.java
package com.example.entity;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Data
@Entity
@Table(name = "doctor_registration")
public class Doctor {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private String doctor_id;
	
	private String firstName;
	
	private String lastName;

	@OneToOne(mappedBy = "users.username")
	private String emailId;
	
	private String specialist;
	
	private String branch;
	
	private String photo;

	private long fee;

	private String password;
}
